File: Matrices/hwMatrix.java
public class Matrix {
	private	int rows, cols;
	private	double[] m;

	// Gauss-Jordan with partial pivoting.  Full pivoting = 200 points
	public double[] solve(double[] B) {
		double x = Double.POSITIVE_INFINITY;
		double w = 1.0e+100;
  }
	public Matrix mult(Matrix b) {


	}
	public void LUFactorize(Matrix L, Matrix U);
	public abstract void gramSchmidt();
}

File: Matrices/hwMatrix.cc
#include <vector>
constexpr double INF = std::numeric_limits<double>::infinity();
class Matrix {
private:
	uint32_t rows, cols;
	double* m;

public:
	// Gauss-Jordan with partial pivoting.  Full pivoting = 200 points
	vector<double> solve(const vector<double> B);
  friend Matrix operator	*(const Matrix& a, const Matrix& b);
	void LUFactorize(Matrix& L, Matrix& U);
	gramSchmidt()
};

/*
	Same for Tridiagonal and Diagonal Matrices

 */

